js/steps.js

```diff
@@ -0,0 +1,143 @@
+    this.CurrentElement.stepContainer.classList.add(classValue);
+    this.CurrentElement.isValid = isValid;
+  }
+  validateStep(isValid) {
+    if (isValid != null || this.validateInOrder) {
+      isValid = isValid === null ? this.defaultValidator : isValid;
+      this.completeStep(isValid);
+    }
+  }
+  isStepInRange(inReverseDirection) {
+    return inReverseDirection ? this.activeStep > 0 : this.activeStep < this.childElements.length - 1;
+  }
+  decrementIncrementActiveStep(inReverseDirection) {
+    this.activeStep += inReverseDirection ? -1 : 1;
+  }
+  validateUntilPoint(point, isValid = this.defaultValidator) {
+    this.activeStep = 0;
+    while (this.activeStep < point) {
+      this.validateStep(isValid);
+      this.activeStep++;
+    }
+  }
+  moveToNextStep(inReverseDirection, isValid) {
+    this.decrementIncrementActiveStep(inReverseDirection);
+    while (this.CurrentElement.type === "substep") {
+      if ((this.validateInOrder && this.CurrentElement.valid == null) || isValid !== null) {
+        this.validateStep(isValid);
+      }
+      this.decrementIncrementActiveStep(inReverseDirection);
+    }
+  }
+  changeStep(inReverseDirection = false, skipSubsteps = true, isValid = null) {
+    this.validateStep(isValid);
+    if (this.isStepInRange(inReverseDirection)) {
+      if (!skipSubsteps) {
+        this.decrementIncrementActiveStep(inReverseDirection);
+      } else {
+        this.moveToNextStep(inReverseDirection, isValid);
+      }
+    }
+    this.changeActiveStep(this.activeStep);
+  }
+  changeToGivenStep(newStep) {
+    this.activeStep = newStep;
+    if (this.validateInOrder) {
+      this.validateUntilPoint(newStep);
+    }
+    this.changeActiveStep(this.activeStep);
+  }
+  changeProgressBarWidth(stepContainer) {
+    this.progressBarContainerDesktop.style.width = `${stepContainer.offsetLeft}px`;
+    this.progressBarContainerMobile.style.height = `${stepContainer.offsetTop}px`;
+  }
+  stepClick(stepObject, stepsFunction) {
+    let stepIndex = stepsFunction.childElements.indexOf(stepObject);
+    stepsFunction.changeToGivenStep(stepIndex);
+  }
+  buildClickListener(stepObject) {
+    stepObject.stepContainer.addEventListener("click", event => {
+      this.stepClick(stepObject, this);
+    });
+  }
+  createEachStepHTMLElements(stepData, parentElement = this.rootElement) {
+    let stepContainer = this.createElementInContainer("div", parentElement);
+    stepContainer.classList.add("step");
+    let labelContainer = this.createElementInContainer("div", stepContainer);
+    labelContainer.classList.add("label");
+    labelContainer.innerHTML = stepData.label;
+    let tooltipContainer = null;
+    if (stepData.tooltip) {
+      tooltipContainer = this.createElementInContainer("div", stepContainer);
+      tooltipContainer.classList.add("tooltip");
+      tooltipContainer.innerHTML = stepData.tooltip;
+    }
+    let restricted = !stepData.restricted ? false : true;
```

Comment:
const instead of let

if stepData.restricted is a boolean in all cases it could be simply restricted = stepData.restricted. If not in all cases it could be restricted = !!stepData.restricted